packages/frontend/src/ee/features/aiCopilot/components/ChatElements/AiChartVisualization.tsx

```diff
@@ -98,6 +105,32 @@ export const AiChartVisualization: FC<Props> = ({
         ],
     );
 
+    const onActiveTabChange = (value: string) => {
+        setActiveTab(activeTabsSchema.parse(value));
+
+        if (
+            value === 'calculation' &&
+            user?.data?.userUuid &&
+            user?.data?.organizationUuid &&
+            agentUuid &&
+            message.threadUuid &&
+            message.uuid
+        ) {
+            track({
+                name: EventName.AI_AGENT_CHART_HOW_ITS_CALCULATED_CLICKED,
+                properties: {
+                    userId: user.data.userUuid,
+                    organizationId: user.data.organizationUuid,
+                    projectId: projectUuid,
+                    aiAgentId: agentUuid,
+                    threadId: message.threadUuid,
+                    messageId: message.uuid,
+                    chartType: chartConfig.type,
+                },
+            });
+        }
+    };
+
```

Comment:
should we keep track (🥁) of this click so we only send it once per visualization? Because you can toggle between views, I guess this will trigger a new event each time a user does that?

Reply:
yup, this is intentional - we can also just do a "unique" count based on the "messageId" in our own lightdash

Reply:
> we can also just do a "unique" count based on the "messageId" in our own lightdash

ah right 😄 I used to be so conscious about the amount of events we trigger because of 💸